src/span_context.js

```diff
@@ -53,19 +53,7 @@ export default class SpanContext {
 
   get traceId(): any {
     if (this._traceId == null && this._traceIdStr != null) {
-      // make sure that the string is 32 or 16 characters long to generate the
-      // corresponding 64 or 128 bits buffer by padding the start with zeros if necessary
-      // https://github.com/jaegertracing/jaeger/issues/1657
-      // At the same time this will enforce that the HEX has an even number of digits, node is expecting 2 HEX characters per byte
-      // https://github.com/nodejs/node/issues/21242
-      const traceIdExactLength = this._traceIdStr.length > 16 ? 32 : 16;
-      if (this._traceIdStr.length === traceIdExactLength) {
-        this._traceId = Utils.newBufferFromHex(this._traceIdStr);
-      } else {
-        const padding = traceIdExactLength === 16 ? '0000000000000000' : '00000000000000000000000000000000';
-        const safeTraceIdStr = (padding + this._traceIdStr).slice(-traceIdExactLength);
-        this._traceId = Utils.newBufferFromHex(safeTraceIdStr);
-      }
+      this._traceId = Utils.newBufferFromHex(this._traceIdStr);
```

Comment:
I don't think I like this part. We have backwards compatibility requirement to parse strings with stripped zeros (potentially coming from other clients). That means `_traceidStr` could be such trimmed ID. If we convert it directly to byte buffer, the buffer could be of random length. I liked the code above that made sure the buffer is always of precisely known length.

Reply:
Do you mean case where I manually set `trace.context().traceIdStr = 'abc'` ?

In other cases `SpanContext.withStringIds()` guarantees valid `traceIdStr`

Reply:
Fair enough. But wouldn't it be better to perform the padding in the constructor, instead of `SpanContext.withStringIds()`?